lib/dialects/mssql/query-generator.js

```diff
@@ -247,14 +249,14 @@ var QueryGenerator = {
     Utils._.forEach(attrValueHashes, function(attrValueHash) {
       // special case for empty objects with primary keys
       var fields = Object.keys(attrValueHash);
-      if (fields.length === 1 && attributes[fields[0]].autoIncrement && attrValueHash[fields[0]] === null) {
+      if (fields.length === 1 && attributes[fields[0]] && attributes[fields[0]].autoIncrement && attrValueHash[fields[0]] === null) {
```

Comment:
This statement has gone a bit complex now. Please extract `attributes[fields[0]]` as a variable (`firstAttr`) so we can simplify the checks. May be we can get rid of `null` check from end as well